### PR TITLE
Refactoring let (e.g. reduce scope)

### PR DESCRIPTION
**Pull request by segayuu**
- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

**Comment by segayuu** [comment]
The cause of failure only with currentVersion (11.0.0) is unknown.

**Comment by tomap** [comment]
Also confused

**Comment by tomap** [comment]
Quick way to fix/debug, split in two your changes, which seem pretty isolated, and make two PRs, and repeat with whichever fails...

**Comment by tomap** [comment]
Oups, sorry, I just spotted that it has nothing to do with your changes. Maybe a minor update on one of our dependency triggered that...

**Comment by segayuu** [comment]
As a result of examination, it turned out that it was caused by the difference of nodejs version.
The initial value of the `depth` option of `util.inspect()` has changed since node v11.
[node v11: `20`](https://nodejs.org/api/util.html#util_util_inspect_object_options) [node v10: `2`](https://nodejs.org/docs/latest-v10.x/api/util.html#util_util_inspect_object_options)
[Here is the test line where the error appears.](https://github.com/hexojs/hexo/blob/3.8.0/test/scripts/helpers/debug.js#L24)
Given the default value change, it is clear that the test on this line will fail on v11.

File: lib/extend/filter.js [hunk 1]
```diff
@@ -70,12 +67,11 @@ Filter.prototype.execSync = function(type, data, options = {}) {
   const filters = this.list(type);
   const ctx = options.context;
   const args = options.args || [];
-  let result;
 
   args.unshift(data);
 
   for (let i = 0, len = filters.length; i < len; i++) {
-    result = filters[i].apply(ctx, args);
+    const result = filters[i].apply(ctx, args);
```
Comment (NoahDragon) [inline lib/extend/filter.js]: I'm not understanding why we need to change this? Due to performance?

Comment (segayuu) [reply lib/extend/filter.js]: If that comment is about this line, the purpose is to reduce the scope of `result`.

**Comment by segayuu** [comment]
An error occurred because the conflict resolution failed. It is resolved now.

**Pull request merged**